qiskit/mapper/_compiling.py

```diff
@@ -334,56 +385,59 @@ def two_qubit_kak(unitary_matrix):
         "name": "cx",
         "args": [1, 0],
         "params": ()
-        })
+    })
     return_circuit.append({
         "name": u1_gate[0],
         "args": [0],
         "params": u1_gate[1]
-        })
+    })
     return_circuit.append({
         "name": u2_gate[0],
         "args": [1],
         "params": u2_gate[1]
-        })
-
-    # Test gate sequence
-    V = np.identity(4)
-    cx21 = np.array([[1, 0, 0, 0],
-                     [0, 0, 0, 1],
-                     [0, 0, 1, 0],
-                     [0, 1, 0, 0]], dtype=complex)
-    cx12 = np.array([[1, 0, 0, 0],
-                     [0, 1, 0, 0],
-                     [0, 0, 0, 1],
-                     [0, 0, 1, 0]], dtype=complex)
-    for gate in return_circuit:
-        if gate["name"] == "cx":
-            if gate["args"] == [0, 1]:
-                V = np.dot(cx12, V)
-            else:
-                V = np.dot(cx21, V)
-        else:
-            if gate["args"] == [0]:
-                V = np.dot(np.kron(rz_array(gate["params"][2]),
-                                   np.identity(2)), V)
-                V = np.dot(np.kron(ry_array(gate["params"][0]),
-                                   np.identity(2)), V)
-                V = np.dot(np.kron(rz_array(gate["params"][1]),
-                                   np.identity(2)), V)
+    })
+
+    if verify_gate_sequence:
```

Comment:
Now you're removing this functionality by default, it would be great if you can completely separate it from the `two_qubit_kak` function. Something like`verify_circuit_equivalence` can work.

Reply:
I think it is actually fine as is.  It is an optional safety check.  Similar checks are done in `scipy.linalg`, for example.

Reply:
We can keep the API if you think so, the functionality could be factored out (as [`scypi.linalg` already does](https://github.com/scipy/scipy/blob/v1.1.0/scipy/linalg/basic.py#L946)).